agent/agent.go

```diff
@@ -658,10 +658,13 @@ func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) error
 
 func (a *Agent) runPostGroupCommands(ctx context.Context, tc *taskContext) {
 	defer a.removeTaskDirectory(tc)
-	defer a.killProcs(ctx, tc, true)
 	if tc.taskConfig == nil {
 		return
 	}
+	// Only killProcs if tc.taskConfig is not nil. This avoids passing an
+	// empty working directory to killProcs, and is okay because this
+	// killProcs is only for the processes run in runPostGroupCommands.
+	defer a.killProcs(ctx, tc, true)
```

Comment:
I think even if we don't do this in runPost GroupCommands, it still makes sense to keep the nil check in killProcs. Is there a reason we aren't concerned about this in the other cases?

Reply:
We've only seen this panic happen in the `runPostGroupCommands` case, but yeah, it does seem like reasonable defensive programming to keep the nil check. I put it back.

Reply:
will it _always_ be what we want to completely skip killProcs when it's nil?

Reply:
I don't know, but I think this is a future code maintainability problem. I explained this in my other comment. We have multiple implementations of `getPIDsToKill`, and I don't think it should be the responsibility of a function to handle odd input, since it puts the burden on the function of understanding what an empty directory means.